onyo/lib/onyo.py

```diff
@@ -203,155 +203,56 @@ def clear_caches(self,
-            An integer specifying the maximal length for generated commit
-            message subjects.
-
-        modified: Iterable of Path, optional
-            A list of Paths to assets/directories modified in the commit, used
-            to generate the commit message subject and body.
-
-        Returns
-        -------
-        str
-            A massage suitable as a commit message for use with `git commit`.
-        """
-        message = [] if message is None else message
-        keys = [] if keys is None else keys
-        if modified is None:
-            modified = self.git.files_staged
-
-        # ensure uniqueness of modified paths
-        modified = list(set(modified))
-
-        message_subject = ""
-
-        # staged files and directories (without ".anchor") in alphabetical order
-        changes_to_record = [x if not x.name == self.ANCHOR_FILE else x.parent
-                             for x in sorted(f.relative_to(self.git.root) for f in modified)]
-
-        if message:
-            message_subject = message[0][0]
-            message_subject += '\n'.join(map(str, [x[0] for x in message[1:]]))
-        else:
-            # get variables for the begin of the commit message `msg_dummy`
-            dest = None
-            keys_str = ""
-            if keys:
-                keys_str = f" ({','.join(str(x.split('=')[0]) for x in sorted(keys))})"
-            if destination:
-                # Note: This seems to expect relative path in `destination`,
-                #       turns it into absolute and back into relative.
-                #       Double-check where `destination` is used and remove
-                #       resolution:
-                dest = (self.git.root / destination).relative_to(self.git.root)
-            if dest and dest.name == self.ANCHOR_FILE:
-                dest = dest.parent
-
-            # the `msg_dummy` is the way all automatically generated commit
-            # message headers (independently of later adding/shortening of
-            # information) begin, for all commands.
-            msg_dummy = f"{cmd} [{len(changes_to_record)}]{keys_str}"
-            message_subject = self._generate_commit_message_subject(
-                msg_dummy, changes_to_record, dest, max_length)
-
-        return f"{message_subject}"
-
-    @staticmethod
-    def _generate_commit_message_subject(msg_dummy: str,
-                                         changes: list[Path],
-                                         destination: Optional[Path],
-                                         max_length: int = 80) -> str:
-        """Generates a "commit message subject" usable with `git commit`.
-        The function lists the paths of `changes` and will shorten the resulting
-        string to try to include as much user readable information as possible.
+        The function will shorten paths in the resulting string to try to include as much
+        user readable information as possible.
```

Comment:
The phrasing is strange. If the goal was to maximize information, we wouldn't shorten anything.
The point is `max_length`.
As for "user readable information": What would the function do differently, if "user readability" was not a concern? Can machines extract more information from shorten paths? ;-) 


```suggestion
        The function will shorten paths in the resulting string in order to try to fit into `max_length`.
```